File: inventory-service/src/main/java/br/com/maccommerce/inventoryservice/domain/exception/ApiException.java
package br.com.maccommerce.inventoryservice.domain.exception;

public class ApiException extends RuntimeException {

    public final ApiExceptionType type;
    public final String message;

    ApiException(ApiExceptionType type, String message) {
        super(message);
        this.type = type;
        this.message = message;
    }

    public enum ApiExceptionType {

        NOT_FOUND,
        INTERNAL_SERVER_ERROR

    }

}

File: product-service/src/main/kotlin/br/com/maccommerce/productservice/domain/exception/ApiException.kt
package br.com.maccommerce.productservice.domain.exception

enum class ApiExceptionType {

    BAD_REQUEST,
    DATABASE_ERROR,
    INTERNAL_ERROR,
    NOT_FOUND

}

open class ApiException(
    val type: ApiExceptionType,
    override val message: String,
    override val cause: Throwable? = null
) : RuntimeException(message)
